File: bankingOnGolang/internal/model/Collateral.go
package model

import ()

// ==============================================================
// Collateral Declaration
// ==============================================================
type Collateral struct {
	BaseModel
	CollateralIdentifier string
	AppraisedValue       Money
	Description          string
	Location             Address
	LoanAccountId        *uint
	LoanAccount          *LoanAccount `gorm:"foreignKey:LoanAccountId"`
	CollateralType       CollateralType

	// parent associations as their child

}

File: src/app/models/Collateral.ts


// Define collection and schema for Collateral
export  Collateral {
    collateralIdentifier: string
    appraisedValue: Money
    description: string
    location: Address
    LoanAccount: Schema.Types.ObjectId
    CollateralType:  String
    collection: 'collaterals'
}
